cloud-optimized-geotiffs/cogs-details.qmd

```diff
@@ -1,28 +1,29 @@
+**Recommendation** The smallest possible data type that still represents the data appropriately should be used. It is not generally recommended to shift or quantize data from float to integer by multiplying, a space saving technique, as end users then need to undo this step to use the data. Data compression is preferred, see also [Compression](#compression).
 
-GeoTIFF format supports many data types. The key is that all bands must be of the same data type. Unlike some other formats you can not mix and match integers (whole numbers) and floats (decimal numbers) in the same file. If you have this use case consider splitting files by data type and using a catalog like STAC to keep track of them, or look at other formats like [Zarr](../zarr/intro.qmd).
+The GeoTIFF format supports many data types. The key is that all bands must be of the same data type. Unlike some other formats, you cannot mix and match integer (whole number) and float (decimal number) data types in the same file. If you have this use case consider splitting files by data type and using a catalog like STAC to keep track of them, or look at other formats like [Zarr](../zarr/intro.qmd).
 
-Scenario: If the COG is intended only for visualization, conversion to 3 band byte will improve performance. 
+Scenario: If the COG is intended only for visualization, conversion to 3 band byte will improve performance.
 
 > GDAL supported Data Types [list](https://gdal.org/drivers/raster/gtiff.html#gtiff-geotiff-file-format)
 
 
 ## Compression (aka File Size)
 
-The biggest benefit to compression is on the storage side. It’s always recommended to use a lossless compression method. **Deflate** or **LZW** are the most recommended compression algorithms, there are some choices that depend on the data type and distribution, and if the goal is maximum compression or not. Maximum compression does result in some performance loss.
+The biggest benefit to compression is on the storage side. It’s always recommended to use a lossless compression method unless the COG is intended to be used only for visualization. **Deflate** or **LZW** are the recommended compression algorithms for general use. **LERC** (Limited Error Raster Compression) provides very efficient data compression for float data if you're ok rounding to some precision you provide. (This could be useful, e.g. for elevation data where the source data is known to not have precision beyond a known value.) Some other compression algorithms may be preferred depending on the data type and distribution, and if the goal is maximum compression or not. Maximum compression does result in some performance loss.
```

Comment:
👍 / 👎 on specifically mentioning LERC? Given that this is the "advanced" page, I think it's appropriate

Reply:
👍 

Maybe mention clearly that LERC is not lossless ;-)

Reply:
Might be good to mention that some compression schemas are optional for GDAL builds so they should be the compression schema should be use wisely

Reply:
@kylebarron what did you think about linking to or citing benchmarks on difference compression settings. Deflate always confuses people, particularly the compression level and predictor.

Reply:
It came up today in the SDSL conference that LERC isn't yet supported in R `terra` builds out of the box https://github.com/rspatial/terra/issues/1205, so I think maybe it would be good to move LERC into a second paragraph where we can explicitly mention that it's lossless and might not be available everywhere.

It would be nice to at least reference some benchmarks; creating our own benchmarks is probably out of scope. The foremost blog post on raster compression I'm aware of is https://kokoalberti.com/articles/geotiff-compression-optimization-guide/. That post also links to http://blog.cleverelephant.ca/2015/02/geotiff-compression-for-dummies.html, https://digital-geography.com/geotiff-compression-comparison/, and https://openterrain.tumblr.com/post/117553678231/gdal-compression-options-against-ned-data

Reply:
I rewrote this section, hopefully improved!